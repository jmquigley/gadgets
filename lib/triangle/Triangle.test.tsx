'use strict';

import {mount, shallow} from 'enzyme';
import * as React from 'react';
import {
	Direction,
	getDefaultTriangleProps,
	Sizing,
	Triangle
} from '../../dist/bundle';

test('Test retrieval of Triangle props object', () => {
	const props = getDefaultTriangleProps();

	expect(props).toBeDefined();
	expect(props).toMatchSnapshot();
});

test('Test creation of a Triangle control', () => {
	const ctl = shallow(<Triangle className="test-class" />);

	expect(ctl).toBeDefined();
	expect(ctl).toMatchSnapshot();
});

for (const direction in Direction) {
	if (Direction.hasOwnProperty(direction)) {

		test(`Create a triangle in ${direction} direction`, () => {
			const ctl = mount(
				<Triangle
					className="test-class"
					direction={direction}
					sizing={Sizing.large}
					style={{
						fill: 'red',
						stroke: 'green'
					}}
				/>
			);

			expect(ctl).toBeDefined();
			expect(ctl).toMatchSnapshot();
		});
	}
}
