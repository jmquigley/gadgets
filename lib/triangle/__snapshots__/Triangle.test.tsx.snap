// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create a triangle in bottom direction 1`] = `
<Triangle
  backgroundColor={null}
  border={null}
  borderColor={null}
  bottom={null}
  className="test-class"
  color={null}
  controlled={true}
  direction="bottom"
  disabled={false}
  err={null}
  errorMessage={null}
  focus={false}
  height={null}
  hidden={false}
  id={null}
  left={null}
  location="none"
  maxHeight={null}
  maxWidth={null}
  minHeight={null}
  minWidth={null}
  nobase={false}
  noborder={false}
  noedit={false}
  nohover={false}
  nopropagation={false}
  notheme={false}
  notooltip={false}
  padding={null}
  ripple={false}
  selected={false}
  sizing="large"
  style={
    Object {
      "fill": "red",
      "stroke": "green",
    }
  }
  testing={true}
  theme={
    Object {
      "backgroundColor": "#ffffff",
      "borderColor": "#c0c0c0",
      "buttonColor": "#666666",
      "chevronColor": "#cccccc",
      "color": "#000000",
      "fontFamily": "OpenSans, Georgia, serif",
      "headerBackgroundColor": "#004358",
      "headerForegroundColor": "#ffffff",
      "headerHoverColor": "#ffe11a",
      "hoverColor": "#c0c0c0",
      "inputBorderColor": "#c0c0c0",
      "itemHoverColor": "#bedb39",
      "outlineColor": "#51cbee",
      "searchFocus": "#51cbee",
      "searchMatch": "#fd7400",
      "selectedBackgroundColor": "#1f8a70",
      "selectedForegroundColor": "#ffe11a",
      "titleBarBackgroundColor": "#004358",
      "titleBarForegroundColor": "#ffe11a",
      "tooltipBackgroundColor": "#708090",
      "tooltipForegroundColor": "#ffffff",
      "transitionDelay": "0.33s",
    }
  }
  tooltip={null}
  top={null}
  visible={true}
  width={null}
  xcss={null}
/>
`;

exports[`Create a triangle in down direction 1`] = `
<Triangle
  backgroundColor={null}
  border={null}
  borderColor={null}
  bottom={null}
  className="test-class"
  color={null}
  controlled={true}
  direction="down"
  disabled={false}
  err={null}
  errorMessage={null}
  focus={false}
  height={null}
  hidden={false}
  id={null}
  left={null}
  location="none"
  maxHeight={null}
  maxWidth={null}
  minHeight={null}
  minWidth={null}
  nobase={false}
  noborder={false}
  noedit={false}
  nohover={false}
  nopropagation={false}
  notheme={false}
  notooltip={false}
  padding={null}
  ripple={false}
  selected={false}
  sizing="large"
  style={
    Object {
      "fill": "red",
      "stroke": "green",
    }
  }
  testing={true}
  theme={
    Object {
      "backgroundColor": "#ffffff",
      "borderColor": "#c0c0c0",
      "buttonColor": "#666666",
      "chevronColor": "#cccccc",
      "color": "#000000",
      "fontFamily": "OpenSans, Georgia, serif",
      "headerBackgroundColor": "#004358",
      "headerForegroundColor": "#ffffff",
      "headerHoverColor": "#ffe11a",
      "hoverColor": "#c0c0c0",
      "inputBorderColor": "#c0c0c0",
      "itemHoverColor": "#bedb39",
      "outlineColor": "#51cbee",
      "searchFocus": "#51cbee",
      "searchMatch": "#fd7400",
      "selectedBackgroundColor": "#1f8a70",
      "selectedForegroundColor": "#ffe11a",
      "titleBarBackgroundColor": "#004358",
      "titleBarForegroundColor": "#ffe11a",
      "tooltipBackgroundColor": "#708090",
      "tooltipForegroundColor": "#ffffff",
      "transitionDelay": "0.33s",
    }
  }
  tooltip={null}
  top={null}
  visible={true}
  width={null}
  xcss={null}
/>
`;

exports[`Create a triangle in left direction 1`] = `
<Triangle
  backgroundColor={null}
  border={null}
  borderColor={null}
  bottom={null}
  className="test-class"
  color={null}
  controlled={true}
  direction="left"
  disabled={false}
  err={null}
  errorMessage={null}
  focus={false}
  height={null}
  hidden={false}
  id={null}
  left={null}
  location="none"
  maxHeight={null}
  maxWidth={null}
  minHeight={null}
  minWidth={null}
  nobase={false}
  noborder={false}
  noedit={false}
  nohover={false}
  nopropagation={false}
  notheme={false}
  notooltip={false}
  padding={null}
  ripple={false}
  selected={false}
  sizing="large"
  style={
    Object {
      "fill": "red",
      "stroke": "green",
    }
  }
  testing={true}
  theme={
    Object {
      "backgroundColor": "#ffffff",
      "borderColor": "#c0c0c0",
      "buttonColor": "#666666",
      "chevronColor": "#cccccc",
      "color": "#000000",
      "fontFamily": "OpenSans, Georgia, serif",
      "headerBackgroundColor": "#004358",
      "headerForegroundColor": "#ffffff",
      "headerHoverColor": "#ffe11a",
      "hoverColor": "#c0c0c0",
      "inputBorderColor": "#c0c0c0",
      "itemHoverColor": "#bedb39",
      "outlineColor": "#51cbee",
      "searchFocus": "#51cbee",
      "searchMatch": "#fd7400",
      "selectedBackgroundColor": "#1f8a70",
      "selectedForegroundColor": "#ffe11a",
      "titleBarBackgroundColor": "#004358",
      "titleBarForegroundColor": "#ffe11a",
      "tooltipBackgroundColor": "#708090",
      "tooltipForegroundColor": "#ffffff",
      "transitionDelay": "0.33s",
    }
  }
  tooltip={null}
  top={null}
  visible={true}
  width={null}
  xcss={null}
/>
`;

exports[`Create a triangle in right direction 1`] = `
<Triangle
  backgroundColor={null}
  border={null}
  borderColor={null}
  bottom={null}
  className="test-class"
  color={null}
  controlled={true}
  direction="right"
  disabled={false}
  err={null}
  errorMessage={null}
  focus={false}
  height={null}
  hidden={false}
  id={null}
  left={null}
  location="none"
  maxHeight={null}
  maxWidth={null}
  minHeight={null}
  minWidth={null}
  nobase={false}
  noborder={false}
  noedit={false}
  nohover={false}
  nopropagation={false}
  notheme={false}
  notooltip={false}
  padding={null}
  ripple={false}
  selected={false}
  sizing="large"
  style={
    Object {
      "fill": "red",
      "stroke": "green",
    }
  }
  testing={true}
  theme={
    Object {
      "backgroundColor": "#ffffff",
      "borderColor": "#c0c0c0",
      "buttonColor": "#666666",
      "chevronColor": "#cccccc",
      "color": "#000000",
      "fontFamily": "OpenSans, Georgia, serif",
      "headerBackgroundColor": "#004358",
      "headerForegroundColor": "#ffffff",
      "headerHoverColor": "#ffe11a",
      "hoverColor": "#c0c0c0",
      "inputBorderColor": "#c0c0c0",
      "itemHoverColor": "#bedb39",
      "outlineColor": "#51cbee",
      "searchFocus": "#51cbee",
      "searchMatch": "#fd7400",
      "selectedBackgroundColor": "#1f8a70",
      "selectedForegroundColor": "#ffe11a",
      "titleBarBackgroundColor": "#004358",
      "titleBarForegroundColor": "#ffe11a",
      "tooltipBackgroundColor": "#708090",
      "tooltipForegroundColor": "#ffffff",
      "transitionDelay": "0.33s",
    }
  }
  tooltip={null}
  top={null}
  visible={true}
  width={null}
  xcss={null}
/>
`;

exports[`Create a triangle in top direction 1`] = `
<Triangle
  backgroundColor={null}
  border={null}
  borderColor={null}
  bottom={null}
  className="test-class"
  color={null}
  controlled={true}
  direction="top"
  disabled={false}
  err={null}
  errorMessage={null}
  focus={false}
  height={null}
  hidden={false}
  id={null}
  left={null}
  location="none"
  maxHeight={null}
  maxWidth={null}
  minHeight={null}
  minWidth={null}
  nobase={false}
  noborder={false}
  noedit={false}
  nohover={false}
  nopropagation={false}
  notheme={false}
  notooltip={false}
  padding={null}
  ripple={false}
  selected={false}
  sizing="large"
  style={
    Object {
      "fill": "red",
      "stroke": "green",
    }
  }
  testing={true}
  theme={
    Object {
      "backgroundColor": "#ffffff",
      "borderColor": "#c0c0c0",
      "buttonColor": "#666666",
      "chevronColor": "#cccccc",
      "color": "#000000",
      "fontFamily": "OpenSans, Georgia, serif",
      "headerBackgroundColor": "#004358",
      "headerForegroundColor": "#ffffff",
      "headerHoverColor": "#ffe11a",
      "hoverColor": "#c0c0c0",
      "inputBorderColor": "#c0c0c0",
      "itemHoverColor": "#bedb39",
      "outlineColor": "#51cbee",
      "searchFocus": "#51cbee",
      "searchMatch": "#fd7400",
      "selectedBackgroundColor": "#1f8a70",
      "selectedForegroundColor": "#ffe11a",
      "titleBarBackgroundColor": "#004358",
      "titleBarForegroundColor": "#ffe11a",
      "tooltipBackgroundColor": "#708090",
      "tooltipForegroundColor": "#ffffff",
      "transitionDelay": "0.33s",
    }
  }
  tooltip={null}
  top={null}
  visible={true}
  width={null}
  xcss={null}
/>
`;

exports[`Create a triangle in up direction 1`] = `
<Triangle
  backgroundColor={null}
  border={null}
  borderColor={null}
  bottom={null}
  className="test-class"
  color={null}
  controlled={true}
  direction="up"
  disabled={false}
  err={null}
  errorMessage={null}
  focus={false}
  height={null}
  hidden={false}
  id={null}
  left={null}
  location="none"
  maxHeight={null}
  maxWidth={null}
  minHeight={null}
  minWidth={null}
  nobase={false}
  noborder={false}
  noedit={false}
  nohover={false}
  nopropagation={false}
  notheme={false}
  notooltip={false}
  padding={null}
  ripple={false}
  selected={false}
  sizing="large"
  style={
    Object {
      "fill": "red",
      "stroke": "green",
    }
  }
  testing={true}
  theme={
    Object {
      "backgroundColor": "#ffffff",
      "borderColor": "#c0c0c0",
      "buttonColor": "#666666",
      "chevronColor": "#cccccc",
      "color": "#000000",
      "fontFamily": "OpenSans, Georgia, serif",
      "headerBackgroundColor": "#004358",
      "headerForegroundColor": "#ffffff",
      "headerHoverColor": "#ffe11a",
      "hoverColor": "#c0c0c0",
      "inputBorderColor": "#c0c0c0",
      "itemHoverColor": "#bedb39",
      "outlineColor": "#51cbee",
      "searchFocus": "#51cbee",
      "searchMatch": "#fd7400",
      "selectedBackgroundColor": "#1f8a70",
      "selectedForegroundColor": "#ffe11a",
      "titleBarBackgroundColor": "#004358",
      "titleBarForegroundColor": "#ffe11a",
      "tooltipBackgroundColor": "#708090",
      "tooltipForegroundColor": "#ffffff",
      "transitionDelay": "0.33s",
    }
  }
  tooltip={null}
  top={null}
  visible={true}
  width={null}
  xcss={null}
/>
`;

exports[`Test creation of a Triangle control 1`] = `
<Wrapper
  backgroundColor={null}
  border={null}
  borderColor={null}
  bottom={null}
  className="test-class"
  color={null}
  controlled={true}
  direction="up"
  disabled={false}
  err={null}
  errorMessage={null}
  focus={false}
  height={null}
  hidden={false}
  id={null}
  left={null}
  location="none"
  maxHeight={null}
  maxWidth={null}
  minHeight={null}
  minWidth={null}
  name="Triangle"
  nobase={false}
  noborder={false}
  noedit={false}
  nohover={false}
  nopropagation={false}
  notheme={false}
  notooltip={false}
  onError={[Function]}
  padding={null}
  reset={false}
  ripple={false}
  selected={false}
  sizing="normal"
  style={
    Object {
      "fill": "black",
      "stokeWidth": "2px",
      "stroke": "black",
    }
  }
  testing={true}
  theme={
    Object {
      "backgroundColor": "#ffffff",
      "borderColor": "#c0c0c0",
      "buttonColor": "#666666",
      "chevronColor": "#cccccc",
      "color": "#000000",
      "fontFamily": "OpenSans, Georgia, serif",
      "headerBackgroundColor": "#004358",
      "headerForegroundColor": "#ffffff",
      "headerHoverColor": "#ffe11a",
      "hoverColor": "#c0c0c0",
      "inputBorderColor": "#c0c0c0",
      "itemHoverColor": "#bedb39",
      "outlineColor": "#51cbee",
      "searchFocus": "#51cbee",
      "searchMatch": "#fd7400",
      "selectedBackgroundColor": "#1f8a70",
      "selectedForegroundColor": "#ffe11a",
      "titleBarBackgroundColor": "#004358",
      "titleBarForegroundColor": "#ffe11a",
      "tooltipBackgroundColor": "#708090",
      "tooltipForegroundColor": "#ffffff",
      "transitionDelay": "0.33s",
    }
  }
  tooltip={null}
  top={null}
  visible={true}
  width={null}
  xcss={null}
>
  <Triangle__SVGView
    backgroundColor={null}
    border={null}
    borderColor={null}
    bottom={null}
    className="ui-triangle test-class"
    color={null}
    controlled={true}
    direction="up"
    disabled={false}
    err={null}
    errorMessage={null}
    focus={false}
    height={null}
    hidden={false}
    id={null}
    left={null}
    location="none"
    maxHeight={null}
    maxWidth={null}
    minHeight={null}
    minWidth={null}
    nobase={false}
    noborder={false}
    noedit={false}
    nohover={false}
    nopropagation={false}
    notheme={false}
    notooltip={false}
    padding={null}
    preserveAspectRatio="xMidYMid meet"
    ripple={false}
    selected={false}
    sizing="normal"
    style={
      Object {
        "fill": "black",
        "stokeWidth": "2px",
        "stroke": "black",
      }
    }
    testing={true}
    theme={
      Object {
        "backgroundColor": "#ffffff",
        "borderColor": "#c0c0c0",
        "buttonColor": "#666666",
        "chevronColor": "#cccccc",
        "color": "#000000",
        "fontFamily": "OpenSans, Georgia, serif",
        "headerBackgroundColor": "#004358",
        "headerForegroundColor": "#ffffff",
        "headerHoverColor": "#ffe11a",
        "hoverColor": "#c0c0c0",
        "inputBorderColor": "#c0c0c0",
        "itemHoverColor": "#bedb39",
        "outlineColor": "#51cbee",
        "searchFocus": "#51cbee",
        "searchMatch": "#fd7400",
        "selectedBackgroundColor": "#1f8a70",
        "selectedForegroundColor": "#ffe11a",
        "titleBarBackgroundColor": "#004358",
        "titleBarForegroundColor": "#ffe11a",
        "tooltipBackgroundColor": "#708090",
        "tooltipForegroundColor": "#ffffff",
        "transitionDelay": "0.33s",
      }
    }
    tooltip={null}
    top={null}
    version="1.1"
    viewBox="0 0 40 40"
    visible={true}
    width={null}
    xcss={null}
    xmlns="http://www.w3.org/2000/svg"
  >
    <polygon
      points="0,35, 20,10 40,35"
    />
  </Triangle__SVGView>
</Wrapper>
`;

exports[`Test retrieval of Triangle props object 1`] = `
Object {
  "backgroundColor": null,
  "border": null,
  "borderColor": null,
  "bottom": null,
  "children": null,
  "className": null,
  "color": null,
  "controlled": true,
  "direction": "up",
  "disabled": false,
  "err": null,
  "errorMessage": null,
  "focus": false,
  "height": null,
  "hidden": false,
  "id": null,
  "left": null,
  "location": "none",
  "maxHeight": null,
  "maxWidth": null,
  "minHeight": null,
  "minWidth": null,
  "nobase": false,
  "noborder": false,
  "noedit": false,
  "nohover": false,
  "nopropagation": false,
  "notheme": false,
  "notooltip": false,
  "padding": null,
  "ripple": false,
  "selected": false,
  "sizing": "normal",
  "style": Object {
    "fill": "black",
    "stokeWidth": "2px",
    "stroke": "black",
  },
  "testing": true,
  "theme": Object {
    "backgroundColor": "#ffffff",
    "borderColor": "#c0c0c0",
    "buttonColor": "#666666",
    "chevronColor": "#cccccc",
    "color": "#000000",
    "fontFamily": "OpenSans, Georgia, serif",
    "headerBackgroundColor": "#004358",
    "headerForegroundColor": "#ffffff",
    "headerHoverColor": "#ffe11a",
    "hoverColor": "#c0c0c0",
    "inputBorderColor": "#c0c0c0",
    "itemHoverColor": "#bedb39",
    "outlineColor": "#51cbee",
    "searchFocus": "#51cbee",
    "searchMatch": "#fd7400",
    "selectedBackgroundColor": "#1f8a70",
    "selectedForegroundColor": "#ffe11a",
    "titleBarBackgroundColor": "#004358",
    "titleBarForegroundColor": "#ffe11a",
    "tooltipBackgroundColor": "#708090",
    "tooltipForegroundColor": "#ffffff",
    "transitionDelay": "0.33s",
  },
  "tooltip": null,
  "top": null,
  "visible": true,
  "width": null,
  "xcss": null,
}
`;

exports[`Test retrieval of Triangle props object 2`] = `
Object {
  "children": null,
  "sizing": "normal",
  "style": Object {},
}
`;
